pkg/git/poll/gitrepopolljob.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021-2024 SUSE LLC
+package poll
+
+import (
+	"context"
+	"fmt"
+
+	v1alpha1 "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/reugn/go-quartz/quartz"
+	"golang.org/x/sync/semaphore"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ quartz.Job = &GitRepoPollJob{}
+
+type GitFetcher interface {
+	LatestCommit(ctx context.Context, gitrepo *v1alpha1.GitRepo, client client.Client) (string, error)
+}
+
+type GitRepoPollJob struct {
+	sem     *semaphore.Weighted
+	client  client.Client
+	GitRepo v1alpha1.GitRepo
+	fetcher GitFetcher
+}
+
+func gitRepoPollDescription(gitRepo v1alpha1.GitRepo) string {
```

Comment:
I wonder if this couldn't just be `String() string` and return that in the Description?

Reply:
yup, done.